pkg/extension/sumologicextension/extension.go

```diff
@@ -30,17 +30,21 @@ import (
 	"sync"
 	"time"
 
+	"github.com/google/uuid"
+
 	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/sumologicextension/api"
 	"go.opentelemetry.io/collector/component"
 	"go.uber.org/zap"
 )
 
 type SumologicExtension struct {
+	collectorName     string
```

Comment:
As suggested in the `pkg/extension/sumologicextension/api/register.go`: let's not add this here as it's not obvious what it's being used for: we already have the config? Why add another `collectorName` in the extension struct itself? (those might be the questions that someone reading this for the first time might be asking)

Reply:
This is added to avoid overriding user configuration.